src/dataset/mod.rs

```diff
@@ -323,6 +323,18 @@ pub trait Labels {
     fn labels(&self) -> Vec<Self::Elem> {
         self.label_set().into_iter().flatten().collect()
     }
+
+    fn combined_labels(&self, other: Vec<Self::Elem>) -> Vec<Self::Elem> {
```

Comment:
Better to have this method take `&impl Labels` or `&Self` as input. Then you can call `label_set` on both `self` and the input and union all the hashsets before converting it into a `Vec`.